### DISCUSSION TITLE
Remove older versions of JavaScript library

### DISCUSSION THREAD

**Discussion by rcgeekhub** [dnnsoftware/Dnn.Platform#4785] [Q&A]
Hi all, we have upgraded our DNN. Here is our upgrade path DNN 9.3.2 > 9.6.1 > 9.9.1

We realised that the older versions of JS library still remained (e.g. jQuery 3.2.1) and we wish to remove.
How do we do that without breaking the system?

**Comment by valadas** [comment]
In theory, that should have been upgraded and not replaced. When you say it remained, is it listed in the extensions or are you just talking about the files?

If we are talking about only the files, they should be safe to remove but don't forget to backup first. I would test any third-party module after that just in case after too.

**Reply by bdukes** [reply]
Since JavaScript Library extensions support multiple installed versions, installing a new version of, e.g. jQuery, does not upgrade the existing version but adds a new version (i.e. leaving the old version as well). 

And I think the heart of the issue is that there are JavaScript Libraries marked as System libraries, which means they can't be uninstalled, but they probably should be able to be uninstalled if there's a newer version.

It looks like an uninstall step was added to 9.2.1, so maybe we need to add another one of those: https://github.com/dnnsoftware/Dnn.Platform/blob/de333bfe6dc13a132f8dcde6f2c366920e75f612/DNN%20Platform/Library/Services/Upgrade/Upgrade.cs#L2992-L2997

I think if you update the `IsSystem` column for the JS library's package, you should be able to delete it through the UI (I feel like there was maybe one other step, but I can't find where I've documented the steps previously)

**Comment by rcgeekhub** [comment]
If it is not possible to remove, how to delink them especially if there are some vulnerabilities with the older versions? 

Below is what i am seeing. Is it only the host can update these library settings? can the admin update?
![dnn](https://user-images.githubusercontent.com/66811409/129657958-daed6442-2b2a-4853-b1f3-55cfaa8e6efc.jpg)

**Reply by rcgeekhub** [reply]
We managed to remove the older version of JS library, but encountered problem removing older version of DLL files. The system didn't render the pages as per usual.
